### DISCUSSION TITLE
How to don't display console window on Mac OS and Linux

### DISCUSSION THREAD

**Discussion by serhii-tokariev84** [hecrj/iced#1144] [Q&A] [answered]
When running application on Mac OS the console window opens aside with app window.

For Windows that could be fixed by `#![windows_subsystem = "windows"]`. How to fix that for Mac OS and Linux?

**Comment by fenhl** [comment, marked as answer]
How are you launching your app? It might be worth a try to check whether [bundling an .app](https://stackoverflow.com/a/65393488/667338) fixes this.

**Reply by serhii-tokariev84** [reply]
Yes, that helped, thanks a lot

**Comment by manokara** [comment]
That wouldn't happen on Linux given how console access works, and I think macOS being BSD-like probably works the same way, but it might be checking if the app is accessing the stdout file descriptor and opens a console window automatically. In that case, if there was a system function similar to [`GetConsoleWindow`](https://docs.microsoft.com/en-us/windows/console/getconsolewindow) for macOS you could close it when the app starts?

**Reply by serhii-tokariev84** [reply]
Thanks, will try to remove all calls of print!

**Reply by serhii-tokariev84** [reply]
Didn't help, really. Removed all console output, console window still shown.
Seems like app are run via shell.
